frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/pipeline-details/pipeline-topology/factories.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  ComponentFactory,
+  DagreLayout,
+  GraphComponent,
+  LayoutFactory,
+  ModelKind,
+  Graph,
+} from '@console/topology';
+import { LayoutCallback } from '@console/topology/src/layouts/DagreLayout';
+import {
+  NODE_SEPARATION_HORIZONTAL,
+  NodeType,
+  PipelineLayout,
+  NODE_SEPARATION_VERTICAL,
+} from './const';
+import SpacerNode from './SpacerNode';
+import TaskNode from './TaskNode';
+import TaskEdge from './TaskEdge';
+
+export const componentFactory: ComponentFactory = (kind: ModelKind, type: string) => {
+  switch (kind) {
+    case ModelKind.graph:
+      return GraphComponent;
+    case ModelKind.edge:
+      return TaskEdge;
+    case ModelKind.node:
+      switch (type) {
+        case NodeType.TASK_NODE:
+          return TaskNode;
+        case NodeType.SPACER_NODE:
+          return SpacerNode;
+        default:
+          return undefined;
+      }
+    default:
+      return undefined;
+  }
+};
+
+// TODO: Fix this hack as it's not the best way to get the layout update
+type CallbackLayout = (onLayout: LayoutCallback) => LayoutFactory;
```

Comment:
@jeff-phillips-18 This is the hack we discussed - we need to find a better way for me to get a notification when the layout finishes.